x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/NumericAggregate.java

```diff
@@ -36,14 +36,20 @@ protected TypeResolution resolveType() {
         if (supportsDates()) {
             return TypeResolutions.isType(
                 this,
-                e -> e.isNumeric() || e == DataTypes.DATETIME,
+                e -> e.isNumeric() && e != DataTypes.UNSIGNED_LONG || e == DataTypes.DATETIME,
                 sourceText(),
                 DEFAULT,
-                "numeric",
+                "numeric except unsigned_long",
```

Comment:
Here I would use `numeric except unsigned_long` **after** `datetime`. The error message is better visualized as 
`argument of [max(x)] must be [datetime or numeric except unsigned_long]`
rather than
`argument of [max(x)] must be [numeric except unsigned_long or datetime]`

Reply:
+1, it's definitely more readable